src/main/java/com/iridium/iridiumskyblock/utils/PlayerUtils.java

```diff
@@ -1,26 +1,40 @@
 package com.iridium.iridiumskyblock.utils;
 
-import com.earth2me.essentials.Essentials;
-import com.earth2me.essentials.spawn.EssentialsSpawn;
 import com.iridium.iridiumskyblock.IridiumSkyblock;
+import com.iridium.iridiumskyblock.database.Island;
+import com.iridium.iridiumteams.support.SpawnSupport;
+
 import org.bukkit.Bukkit;
+import org.bukkit.Location;
 import org.bukkit.World;
 import org.bukkit.entity.Player;
 
+import java.util.Optional;
+
 public class PlayerUtils {
+
     public static void teleportSpawn(Player player) {
-        World spawnWorld = Bukkit.getWorld(IridiumSkyblock.getInstance().getConfiguration().spawnWorldName);
-        if (spawnWorld == null) {
-            spawnWorld = Bukkit.getWorlds().get(0);
+        player.teleport(getSpawn(player));
+    }
+
+    public static Location getSpawn(Player player) {
+
+        if(IridiumSkyblock.getInstance().getConfiguration().spawnOnIsland) {
+            Optional<Island> island = IridiumSkyblock.getInstance().getTeamManager().getTeamViaNameOrPlayer(player.getName());
+            if(island.isPresent()) {
+                return island.get().getHome();
+            }
         }
```

Comment:
iirc, this method is used to find a place to teleport a player when their island gets deleted. so not sure if this will break stuff

Reply:
since we're using an optional here, if it does not return an island (which it shouldnt since the team was deleted), it should teleport the player to the spawn point

Reply:
It depends if we delete the island then teleport them, or teleport them and then delete the island... lemme check in code

Reply:
kk